src/domain/relay/limit-addresses.mapper.ts

```diff
@@ -0,0 +1,74 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { SafeContractHelper } from './contracts/safe-contract.helper';
+import { Erc20ContractHelper } from './contracts/erc20-contract.helper';
+import { Hex } from 'viem/src/types/misc';
+import { ILoggingService, LoggingService } from '@/logging/logging.interface';
+
+export interface RelayPayload {
+  chainId: string;
+  data: Hex;
+  to: Hex;
+  gasLimit: bigint;
+}
+
+@Injectable()
+export class LimitAddressesMapper {
+  constructor(
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+    private readonly safeContract: SafeContractHelper,
+    private readonly erc20Contract: Erc20ContractHelper,
+  ) {}
+
+  getLimitAddresses(relayPayload: RelayPayload): Hex[] {
+    if (this.isValidExecTransactionCall(relayPayload.to, relayPayload.data)) {
+      return [relayPayload.to];
+    }
+
+    // TODO Handle Multisend
+
+    // TODO Handle create proxy with nonce
+
+    throw Error('Cannot get limit addresses – Invalid transfer');
+  }
+
+  private isValidExecTransactionCall(to: string, data: Hex): boolean {
+    let execTransaction: { data: Hex; to: Hex; value: bigint };
+    // If transaction is an execTransaction
+    try {
+      execTransaction = this.safeContract.decode(
+        SafeContractHelper.EXEC_TRANSACTION,
+        data,
+      );
+    } catch (e) {
+      return false;
+    }
+
+    // If data of execTransaction is an ERC20 transfer
+    try {
+      const erc20DecodedData = this.erc20Contract.decode(
+        Erc20ContractHelper.TRANSFER,
+        execTransaction.data,
+      );
+      // If the ERC20 transfer targets 'self' (the Safe), we consider it to be invalid
+      return erc20DecodedData.to !== to;
+    } catch {
+      this.loggingService.debug(
+        'execTransaction data is not an ERC20 transfer',
+      );
```

Comment:
I think this log is excessive. Any Safe method that isn't `execTransaction` will trigger this.

Reply:
Done [4f16ddf](https://github.com/safe-global/safe-client-gateway/pull/680/commits/4f16ddf9e78c9526f8f057773493beebcc2a9c67).

But this log was triggered if an `execTransaction` didn't have an ERC20 transfer 😉